### PR TITLE
helpers for datadict

### PR DESCRIPTION
**Pull request by wpfff**
- avoid deepcopy
- meta-object for easier data access

### CODE REVIEW

**Comment by wpfff** [comment]
@marcosfrenkel can you have a look?
Also -- where should the new data access object be documented?

two main changes:
- deepcopying was in many places, and it's very redundant (we do it anyway in all plottr nodes). so i removed copying as default from pretty much anywhere in datadict, except where you'd really want it (like when returning the structure of the data)
- there's a new attribute to datadict: data of field 'x' can now be accessed as `data.d_.x` -- this gives direct access to the values of field 'x'.

File: plottr/data/datadict.py [hunk 1]
```diff
@@ -618,7 +624,7 @@ def remove_unused_axes(self: T) -> T:
         """
         dependents = self.dependents()
         unused = []
-        ret = self.copy()
+        ret = self  #.copy()
```
Comment (marcosfrenkel) [inline plottr/data/datadict.py]: why do you self in a variable and return it, instead of just using self?

**Review by marcosfrenkel — COMMENTED** [review]
Everything other than comment looks good, and even the comment is nothing too important.

**Comment by marcosfrenkel** [comment]
@wpfff, the documentation for datadicts currently live [here](https://github.com/toolsforexperiments/tfe-manual/blob/main/docs/source/plottr/data.rst) and that is the repository where the new feature should be documented. I can update all of the sub-modules there so that the API sections of stuff shows the current code.

File: plottr/data/datadict.py (hunk 1 above)
Comment (wpfff) [reply plottr/data/datadict.py]: fixed -- ready to approve the PR?

**Comment by wpfff** [comment]
@marcosfrenkel i fixed the one thing you mentioned. ready to approve?

**Review by marcosfrenkel — APPROVED** [review]
Looks good!

**Pull request merged**